### DISCUSSION TITLE
Visuelle Darstellung von Merge- und QuickSort

### DISCUSSION THREAD

**Discussion by Maxman002** [AuDojo/AuDojo#30] [Ideas]
## Problem
Damit das frontend die Sortieralgorithmen möglichst gut visuell darstellen kann, braucht dieses teilweise vermutlich mehr Informationen als nur die Liste nach jedem Sortierschritt.

## MergeSort
Bei MergeSort wird benötigt, an welchem Index die Liste geteilt wurde

## Quicksort
Bei QuickSort wird benötigt welches pivot-element verwendet wurde.

## Frage
Sollen diese Informationen mit den einzelnen Listen vom backend übermittelt werden z.B.
```
{
   "processList": [
      {
         "list": [startList], 
         "pivot": null
      },
      {
         "list": [sortStep1], 
         "pivot": 9
      },
      {
         "list": [sortStep2], 
         "pivot": 5
      }, ...
   ]
}
```

oder soll das im frontend intern gelöst werden?

**Comment by anomatomato** [comment]
- Bei MergeSort Teilung Schritt für Schritt mit unterschiedlichen Farben darstellen
- Oder mit Schieben von Gruppen nach oben, dann beim mergen wieder nach unten
